### ISSUE TITLE
MarkLogic returns no results when querying nested data

### ISSUE THREAD

**Issue by damonLL** [quasar-analytics/quasar#1699]
This query returns no results, but the documents have been verified to exist.

```
select first_name, last_name from patients where codes[*].code = "S72.065C"
```

The `explain` output for that query in the REPL is attached.

[explain-nested-query.txt](https://github.com/quasar-analytics/quasar/files/588181/explain-nested-query.txt)

**Comment by wemrysi** [comment]
Thanks for the report, I've confirmed on current master that this is still an issue, likely due to joins.

**Comment by wemrysi** [comment]
This was fixed via #1912 and testing this on quasar `master` shows it is working properly for both `XML` and `JSON` documents in MarkLogic.
